cmd/thanos/query.go

```diff
@@ -36,30 +36,29 @@ func registerQuery(m map[string]setupFunc, app *kingpin.Application, name string
 	maxConcurrentQueries := cmd.Flag("query.max-concurrent", "maximum number of queries processed concurrently by query node").
 		Default("20").Int()
 
-	peers := cmd.Flag("cluster.peers", "initial peers to join the cluster").Strings()
+	peers := cmd.Flag("cluster.peers", "initial peers to join the cluster. It can be either <ip:port>, or <domain:port>.").Strings()
 
-	clusterBindAddr := cmd.Flag("cluster.address", "listen address for clutser").
+	clusterBindAddr := cmd.Flag("cluster.address", "listen address for cluster").
 		Default(defaultClusterAddr).String()
 
 	clusterAdvertiseAddr := cmd.Flag("cluster.advertise-address", "explicit address to advertise in cluster").
 		String()
 
-	m[name] = func(logger log.Logger, metrics *prometheus.Registry) (okgroup.Group, error) {
-		peer, err := joinCluster(
+	m[name] = func(logger log.Logger, req *prometheus.Registry) (okgroup.Group, error) {
+		joinConfig, err := createJoinConfig(
 			logger,
-			cluster.PeerTypeQuery,
 			*clusterBindAddr,
-			*clusterAdvertiseAddr,
-			*apiAddr,
+			os.ExpandEnv(*clusterAdvertiseAddr),
```

Comment:
Is this for the `POD_IP` envvar? I'd have expected that to be subtituted already by whatever starts the process, no? It's rather awkward semantics to have this be done process-internal to some flags.

Reply:
Nope it is not substituted - that's the problem ):